doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,255 @@
+    the input features, usually leading to better generalization. However, higher
+    ``n_hidden`` value (1) would require ``C`` and ``weight_scale`` to be adjusted
+    to avoid potential overfitting, and (2) would increase memory usage, 
+    training time, and prediction time.
+
+  * **Setting C**: if the dataset has few samples or if the samples are noisy, 
+    decreasing ``C`` will increase regularization which in turn reduces 
+    overfitting and might improve generalization. Increasing ``C`` will 
+    therefore decrease regularization.
+
+  * **Setting weight_scale**:  this controls regularization of the weights 
+    the same way as ``C`` does; but instead of regularizing the hidden-to-output 
+    layer weights, it regularizes the input-to-hidden layer weights.
+
+  * **Setting class_weight**: if the dataset contains an imbalanced ratio 
+    of class fequencies, it is desirable to set `class_weight='auto'`, as
+    it assigns more weight to the class having less number of samples.
+
+  * **Setting batch_size**: training on large datasets or having a large 
+    ``n_hidden`` value could cause issues for limited memory. In this case, it is 
+    recommended to set ``batch_size=1000`` in order to process the data in chunks. 
+    This will run the internal sequential learner which avoids computing the non-linear 
+    feature expansion on the full training set at once. Please note that the end 
+    results are the same regardless of the ``batch_size`` value but smaller values could 
+    increase training time.
+
+  * **Preprocessing**: Making sure that the input features vary on the same scale is 
+    important for having consistent prediction results. ``StandardScaler`` and ``**MinMaxScaler``
+    are two appropriate scalers for this task (see :ref:`link <preprocessing_scaler>`). 
+    Also, a good practice is to use ``Pipeline`` as demonstrated in 
+    :ref:`plot_elm_hyperparameters.py <example_neural_networks_plot_elm_hyperparameters.py>`. 
+
+Please see the example below to observe the impact of ``C`` and ``weight_scale`` 
+on testing score.
+
+.. figure:: ../auto_examples/neural_networks/images/plot_elm_hyperparameters_001.png
+   :target: ../auto_examples/neural_networks/plot_elm_hyperparameters.html
+   :align: center
+   :scale: 100%
+
+.. topic:: Examples:
+
+   * :ref:`example_neural_networks_plot_elm_hyperparameters.py`
+
+
+Mathematical formulation
+========================
+
+A standard ELM trains a single-hidden layer feedforward network using the 
+following function,
+
+:math:`y = \beta \cdot \sigma(W^T \cdot X + b)`
+
+where matrices ``X`` and ``y`` represent the input samples and target values, 
+respectively; matrices ``W`` and ``b`` are randomly generated based on a 
+uniform distribution; matrix :math:`\beta` contains unknown variables;
+and :math:`\sigma` is the non-linear, component-wise activation function.
+
+ELM solves for :math:`\beta` using the ridge regression implementation, given
```

Comment:
I think it should say "using ridge regression". The word implementation seems out-of-place in the math section.